### PR TITLE
Parameterize size and significance metrics, improve robustness (SCP-5241)

### PR DESCRIPTION
**Pull request by eweitz**
This helps enable authors to specify a wider variety of size and significance metrics for their differential expression data.

It also improves robustness for delimiter detection, and error handling for formats that aren't yet implemented.  It integrates with the Core updates over in https://github.com/broadinstitute/single_cell_portal_core/pull/1874.

Automated tests have been updated to verify these changes.  To manually test:
* Run: 
> python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 ingest_differential_expression --annotation-name General_Celltype --annotation-type group --annotation-scope study --cluster-name cluster_umap_txt --study-accession SCPdev --ingest-differential-expression --differential-expression-file ../tests/data/author_de/pval_lfc_pvaladj_seurat-like.tsv --method wilcoxon --size-metric avg_log2FC --significance-metric p_val_adj
* Run:
> head -n 2 cluster_umap_txt--General_Celltype--CSN1S1_macrophages--eosinophils--study--wilcoxon.tsv
* Confirm output:
```
	genes	avg_log2FC	p_val_adj	pct.2	pct.1	p_val	cluster
0	ACE2	0.02746239327	0.1815101221	0.6656237486	0.3343762514	0.1815101221	3
```

This satisfies SCP-5241.

### CODE REVIEW

File: ingest/author_de.py [hunk 1]
```diff
@@ -306,42 +362,53 @@ def __init__(
...
-        # TODO:
-        #   - Throw well-formatted error if file type not in ALLOWED_FILE_TYPES
-        #   - Consider if / how this merges with centralized handling in ingest_files.py
-        file_type = IngestFiles.get_file_type(file_path)[0]
-        if file_type == 'text/tab-separated-values':
-            delimiter = '\t'
-        else:
-            delimiter = ','
-        data = pd.read_csv(file_path, delimiter)
+        # sep=None invokes detecting separator via csv.Sniffer, per
```
Comment (bistline) [inline ingest/author_de.py:372]: Very nice

**Review by bistline — APPROVED** [review]
Looks good - do we have associated tickets to update the author DE forms & Rails/Life Sciences API integration to collect and pass these values?

**Comment by eweitz** [comment]
> do we have associated tickets to update the author DE forms & Rails/Life Sciences API integration to collect and pass these values?

Good question -- I'm handling that as part of the ticket for https://github.com/broadinstitute/single_cell_portal_core/pull/1874/files.  I just got a basic Core-Ingest integration passing.

**Comment by bistline** [comment]
Seeing the Codecov complaint makes me think we should apply the same logic from https://github.com/broadinstitute/single_cell_portal_core/pull/1709 to this repository.

**Comment by eweitz** [comment]
Good call -- [done](https://github.com/broadinstitute/scp-ingest-pipeline/pull/321/commits/d8af4dcb527521a1b8e7af9ba7a7b6055053ea3d)!  Having the same coverage checks across repos makes sense.

**Review by jlchang — APPROVED** [review]
Manual test performs as described. Minor nit, suggestion to clarify comment for future me.

Approving :)

File: ingest/author_de.py [hunk 2]
```diff
@@ -167,56 +181,77 @@ def sort_comparison_metrics(comparison_metrics):
     # Sort alphabetically
     comparison_metrics = sorted(comparison_metrics)
 
-    # Put qval first
+    # Rank significance 1st (ultimately ranked 2nd)
```
[the remark this answers is not in the document]
Comment (eweitz) [reply ingest/author_de.py]: Done in ede5bf3!

File: ingest/author_de.py [hunk 3]
```diff
@@ -167,56 +181,77 @@ def sort_comparison_metrics(comparison_metrics):
...
     comparison_metrics = sorted(
         comparison_metrics,
-        key=lambda x: x.split('--')[-1] == 'qval',
-        reverse=True
+        key=lambda x: x.split('--')[-1] == significance
     )
 
-    # Put logfoldchanges first
+    # Rank size 1st (ultimately ranked 1st)
```
[the remark this answers is not in the document]
Comment (eweitz) [reply ingest/author_de.py]: Done: ede5bf3

**Pull request merged**